internal/backends/sqlite/metadata/registry.go

```diff
@@ -147,22 +147,25 @@ func (r *Registry) CollectionCreate(ctx context.Context, dbName string, collecti
 
 	tableName := r.CollectionToTable(collectionName)
 
-	// TODO use transactions
-	// https://github.com/FerretDB/FerretDB/issues/2747
-
-	query := fmt.Sprintf("CREATE TABLE %q (_ferretdb_sjson TEXT)", tableName)
-	if _, err = db.ExecContext(ctx, query); err != nil {
-		var e *sqlite.Error
-		if errors.As(err, &e) && e.Code() == sqlitelib.SQLITE_ERROR {
-			return false, nil
+	err = pool.InTransaction(ctx, db, func(tx *sql.Tx) error {
+		query := fmt.Sprintf("CREATE TABLE %q (_ferretdb_sjson TEXT)", tableName)
+		if _, err = tx.ExecContext(ctx, query); err != nil {
+			var e *sqlite.Error
+			if errors.As(err, &e) && e.Code() == sqlitelib.SQLITE_ERROR {
+				return nil
```

Comment:
Could add a comment  that `SQLITE_ERROR` is returned when a table exist? It's generic error code so a bit hard to figure out what is trying to capture. :hugs:

Reply:
done

Reply:
Great, very clear!